addons/website_event_booth_sale/models/sale_order.py

```diff
@@ -42,9 +42,29 @@ def _website_product_id_change(self, order_id, product_id, qty=0, **kwargs):
                     event_booth_registrations_command = new_registrations_commands
                 values['event_booth_registration_ids'] = event_booth_registrations_command
 
+            discount = 0
+            order = self.env['sale.order'].sudo().browse(order_id)
+            if order.pricelist_id.discount_policy == 'without_discount':
+                price_unit = sum(booth.booth_category_id.price for booth in booths)
+                price_reduce = sum(booth.booth_category_id.price_reduce for booth in booths)
+                company_currency = order.company_id.currency_id
+                pricelist_currency = order.pricelist_id.currency_id
+                if company_currency != pricelist_currency:
+                    price_unit = company_currency._convert(
+                        price_unit,
+                        pricelist_currency,
+                        order.company_id,
+                        order.date_order or fields.Datetime.now()
+                    )
```

Comment:
Why is the currency conversion only applied when the pricelist is configured to show the discount ?
If we consider the event booth price to be in the company currency, then it must be converted to the pricelist currency regardless of the pricelist discount_policy.

(Same remark for website_event_sale btw)

Also, what about multi-company? AFAIK, the event scope and records are not strictly restricted to a given company.  Does it really make sense to always consider the price to be specified in the order company currency ?
(In master, it could make sense to apply the same behavior as the product currency_id field, using a fixed "main company" currency for the events/tickets/booths shared between companies :thinking:.)

Reply:
I do agree the flow seems a bit weird, but I tried to not alter alter the code base as much as I can. 
From my understanding, the `ticket.price` and the `booth.price` are computed with the `product.list_price` (which is the not converted base price). The `ticket.price_reduce` and the `booth.price_reduce` are computed with the `product.price` (which is converted and discounted).

In our case, when we are in the `with_discount` situation we only use the `[ticket booth].price_reduce` which is therefore not a problem.

For the the `without_discount`, the price_unit is based on the `[ticket booth].price`. This is why we need to convert this price to compare it to the `price_reduce` and to be able to get the `discount` from it.

Reply:
price_reduce fields computation only uses the `product.price` to apply the pricelist discount to the ticket/booth price, but the resulting amount is still specified in the same currency (and needs to be converted).